hs_model_program/static/js/model_program.js

```diff
@@ -0,0 +1,165 @@
+/**
+ * Created by tonycastronova on 10/12/15.
+ */
+
+
+$(document).ready(function () {
+
+    // check to see if the view is in edit mode
+    if(document.getElementById("id-mpmetadata") != null)
+        // load jquery generated elements
+        loadWidgets();
+
+        // initialize datepicker
+        $( "#modelReleaseDate_picker" ).datepicker({
+            // bind a table view to the on close event
+            onSelect: function (event) {
+                onChange_releaseDate(event);
+            }
+        });
+});
+
+/**
+ * Sets the hidden date field base on the datepicker.  The hidden field is what is submitted.
+ */
+function onChange_releaseDate(event){
+    var date = document.getElementById('id_modelReleaseDate');
+    var date_picker = document.getElementById("modelReleaseDate_picker");
+    date.value = date_picker.value;
+
+    // activate the save button
+    $("#resourceSpecificTab").find('.btn-primary').show();
+}
+
+/**
+ * Builds jQuery multiselect listboxes
+ */
+function loadWidgets(){
+
+    // get the resource id from its url
+    var url = document.URL;
+    var url_array = url.split('/');
+    var shortid = url_array[url_array.length - 2];
+
+    // call django view to get model program metadata files
+    $.ajax({
+        type: "GET",
+        url: '/hsapi/_internal/get-model-metadata/',
+        data: {resource_id: shortid},
+        success: function (data) {
+            // get the fieldset items
+            var fs = document.getElementById("id-mpmetadata").getElementsByTagName('fieldset')[0];
+
+            // get the multiselect items
+            var multiselect = fs.getElementsByClassName('div-multi-select');
+
+            for (var i = 0; i < multiselect.length; i++) {
+
+                // get the metadata term associated with this multiselect field
+                var term = multiselect[i].getAttribute('parent_metadata');
+
+                // get the selected items that are saved to the resource
+                var selected_terms = data[term];
+
+                // loop through the options in the select list
+                var options = $(multiselect[i]).find('option');
+                for (var j = 0; j < options.length; j++) {
+
+                    // set the selected attribute
+                    if ($.inArray(options[j].value, selected_terms) >= 0) {
+                        $(options[j]).attr("selected", 1);
+                    }
+                }
+            }
+
+            // set the initial value for the date picker
+            var date = new Date(data['date_released']);
+            if (isNaN(date) == false)
+                $('#modelReleaseDate_picker').datepicker('setDate', date);
+
+        },
+        error: function (data) {
+            console.log('There was an error with model instance GET.')
```

Comment:
Wouldn't it be better displaying an error message in the page rather than outputting to the browser's console?